### PR TITLE
create annoucement functionality

### LINKED ISSUE #10: Create annoucement functionality

### PR DESCRIPTION
**Pull request by lzawistowsky**

### CODE REVIEW

File: graphql/resolvers/annoucement.js [hunk 1]
```diff
@@ -18,6 +20,45 @@ const annoucement = async () => {
...
+    const user = await User.findOne()
+    if(!user) {
+        const error = new Error('Invalid user')
+        error.code = 401;
+        throw error; 
+    }
+
+    const category = await Category.findById(annoucementInput.category)
+    if(!category) {
+        const error = new Error('Invalid user')
```
Comment (zawolanski) [inline graphql/resolvers/annoucement.js]: Invalid error message.

File: graphql/resolvers/annoucement.js [hunk 2]
```diff
@@ -18,6 +20,45 @@ const annoucement = async () => {
   }
 };
 
+const createAnnoucement = async ({ annoucementInput }) => {
+  try {
+    const user = await User.findOne()
+    if(!user) {
```
Comment (zawolanski) [inline graphql/resolvers/annoucement.js]: You should check if user who wants to create new annoucement is logged in using code like below:
```
if (!isAuth && !userId) {
    const error = new Error('Not authorized');
    error.code = 401;
    throw error;
  }
```
Then you should pass `userId` extracted from token in `auth` middleware to `findOne` (or `findById`) method.

File: graphql/schema/annoucement.js [hunk 3]
```diff
@@ -19,6 +19,22 @@ module.exports = {
...
+
+    input AnnoucementInputData {
+        title: String
+        description: String
+        location: String
+        phone: String
+        email: String
+        images: [String]
+        costs: costsInput
+        category: ID!
```
Comment (zawolanski) [inline graphql/schema/annoucement.js]: I think that every field except `images` should be required.

File: graphql/resolvers/annoucement.js [hunk 4]
```diff
@@ -18,6 +20,45 @@ const annoucement = async () => {
...
+      description: annoucementInput.description,
+      location: annoucementInput.location,
+      phone: annoucementInput.phone,
+      email: annoucementInput.email,
+      images: annoucementInput.images,
+      costs: annoucementInput.costs,
+      categoryId: category,
+      addedBy: user
+    });
+    console.log(annoucement)
```
Comment (zawolanski) [inline graphql/resolvers/annoucement.js]: This `console.log` is unnecessary.

**Review by zawolanski — CHANGES_REQUESTED** [review]
Overall fine, but there are a few things you need to fix.

I hope that you remember about fields validation.

**Review by zawolanski — APPROVED** [review]

**Review by pawelsobkowski — APPROVED** [review]

**Pull request merged**